test/instruments_previous_test.go

```diff
@@ -6,96 +6,133 @@ import (
-	assert.NotEmpty(t, createResponse.Customer.Id)
-	assert.NotEmpty(t, createResponse.Customer.Name)
-	assert.NotEmpty(t, createResponse.Customer.Email)
-
-	getResponse, err := PreviousApi().Instruments.Get(createResponse.Id)
-	assert.Nil(t, err)
-	assert.NotNil(t, getResponse)
-	assert.Equal(t, instruments.Card, getResponse.Type)
-	assert.NotEmpty(t, getResponse.Id)
-	assert.NotEmpty(t, getResponse.Fingerprint)
-	assert.NotEmpty(t, getResponse.ExpiryMonth)
-	assert.NotEmpty(t, getResponse.ExpiryYear)
-	assert.NotEmpty(t, getResponse.Scheme)
-	assert.NotEmpty(t, getResponse.Last4)
-	assert.NotEmpty(t, getResponse.Bin)
-	assert.NotEmpty(t, getResponse.CardType)
-	assert.NotEmpty(t, getResponse.CardCategory)
-	assert.NotEmpty(t, getResponse.ProductId)
-	assert.NotEmpty(t, getResponse.ProductType)
-	assert.NotEmpty(t, getResponse.Customer)
-	assert.NotEmpty(t, getResponse.Customer.Id)
-	assert.NotEmpty(t, getResponse.Customer.Name)
-	assert.NotEmpty(t, getResponse.Customer.Email)
+
+	cases := []struct {
+		name       string
+		responseId string
+		checker    func(*abc.GetInstrumentResponse, error)
+	}{
+		{
+			name:       "when the request is valid then response is not nil",
+			responseId: createResponse.Id,
+			checker: func(response *abc.GetInstrumentResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+				assert.Equal(t, instruments.Card, response.Type)
+				assert.NotEmpty(t, response.Id)
+				assert.NotEmpty(t, response.Fingerprint)
+				assert.NotEmpty(t, response.ExpiryMonth)
+				assert.NotEmpty(t, response.ExpiryYear)
+				assert.NotEmpty(t, response.Scheme)
+				assert.NotEmpty(t, response.Last4)
+				assert.NotEmpty(t, response.Bin)
+				assert.NotEmpty(t, response.CardType)
+				assert.NotEmpty(t, response.CardCategory)
+				assert.NotEmpty(t, response.ProductId)
+				assert.NotEmpty(t, response.ProductType)
+				assert.NotEmpty(t, response.Customer)
+				assert.NotEmpty(t, response.Customer.Id)
+				assert.NotEmpty(t, response.Customer.Name)
+				assert.NotEmpty(t, response.Customer.Email)
+			},
+		},
+	}
+
+	client := PreviousApi().Instruments
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.checker(client.Get(tc.responseId))
+		})
+	}
 }
 
 func TestCreateAndUpdateInstrumentPrevious(t *testing.T) {
-
 	cardTokenResponse := RequestCardTokenPrevious(t)
 	createResponse := createTokenInstrumentPrevious(t, cardTokenResponse)
-
 	updateRequest := abc.UpdateInstrumentRequest{
 		ExpiryMonth: 12,
 		ExpiryYear:  2026,
 		Name:        "New Name",
 	}
 
-	updateResponse, err := PreviousApi().Instruments.Update(createResponse.Id, updateRequest)
-	assert.Nil(t, err)
-	assert.NotNil(t, updateResponse)
-	assert.Equal(t, instruments.Card, updateResponse.Type)
-	assert.NotEmpty(t, updateResponse.Fingerprint)
+	cases := []struct {
+		name          string
+		responseId    string
+		updateRequest abc.UpdateInstrumentRequest
+		checker       func(*abc.UpdateInstrumentResponse, error)
```

Comment:
Here you could do something similar to what we have in `TestUpdateWorkflowAction` where we use a `putChecker` to validate the response from the updating operation and then we have a `getChecker` to validate that the data was correctly modified. This test should be called `TestUpdateInstrument`

Reply:
👍